plat/common/x86/syscall.S

```diff
@@ -63,6 +65,8 @@ ENTRY(_ukplat_syscall)
 	 */
 	movq	%gs:LCPU_AUXSP_OFFSET, %rsp
 
+	pushq_cfi	$(GDT_DESC_OFFSET(GDT_DESC_DATA))
```

Comment:
> The process taken to reach a certain state during development does not need to be reflected in the commits merged

I have a different opinion here actually. I believe that it is good for the commits and their messages to attempt to tell the story of the author's train of thought and how he has reached this feature: first, there came the stack switch implementation and **then** came the bonus implementation that also fills in these registers to their actual value. They are entirely unrelated IMO, so I would prefer I keep the two commits separate. In fact, this commit does not help that much to be honest, it's really a bonus so that we don't patch a `clone` child's `iretq` required registers to some known values and have them instead be inherited by the parent. That being said, this commit should've probably been part of #1175 instead.